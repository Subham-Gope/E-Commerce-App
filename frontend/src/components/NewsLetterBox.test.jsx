// import { describe, expect, test, vi } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import NewsLetterBox from "./NewsLetterBox";

// describe("Testing the NewsLetterBox.jsx file", () => {
//   test("renders news letter subscribe form", async () => {
//     // setting up userEvent
//     const user = userEvent.setup();

//     //rendering and setting up the snapshot matching
//     const { container } = render(<NewsLetterBox />);
//     expect(container).toMatchSnapshot();

//     // CHecking the description texts
//     const text1 = screen.getByText(/subscribe now & get 20% off/i);
//     expect(text1).toBeInTheDocument();
//     expect(text1).toHaveTextContent(/20% off/i);

//     const text2 = screen.getByText(/limited-time offer!/i);
//     expect(text2).toBeInTheDocument();
//     expect(text2).toHaveTextContent(/donâ€™t miss out on/i);

//     // Initiating user interaction
//     const btn = screen.getByRole("button", { name: /subscribe/i });
//     const input = screen.getByPlaceholderText(/enter your email/i);
//     expect(input).toBeInTheDocument();
//     expect(btn).toBeInTheDocument();

//     // Listening for form submit
//     const form = screen.getByRole("form");
//     expect(form).toBeInTheDocument();

//     const submitListerner = vi.fn((e) => e.preventDefault());
//     form.addEventListener("submit", submitListerner);

//     await user.type(input, "gigachad@contact.com");
//     await user.click(btn);

//     expect(submitListerner).toHaveBeenCalled();
//   });
// });
