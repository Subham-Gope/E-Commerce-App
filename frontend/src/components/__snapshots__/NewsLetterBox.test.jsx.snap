// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the NewsLetterBox.jsx file > renders news letter subscribe form 1`] = `
<div>
  <div
    class="text-center"
  >
    <p
      class="text-2xl font-medium text-gray-800"
    >
      Subscribe now & get 20% off
    </p>
    <p
      class="text-gray-400 mt-3"
    >
      Subscribe now to stay updated and enjoy an exclusive 20% discount on your first purchase. Donâ€™t miss out on this limited-time offer!
    </p>
    <form
      action="https://app.us2.list-manage.com/subscribe/post?u=29f2bf7df8ccaa4d054839728&id=3a4ce84596&f_id=00c1a4e0f0"
      aria-label="form"
      class="w-full sm:w-1/2 flex items-center gap-3 mx-auto my-6 border pl-3 max-w-md"
      method="POST"
      target="_blank"
    >
      <input
        class="w-full sm:flex-1 outline-none"
        name="EMAIL"
        placeholder="Enter your email"
        required=""
        type="email"
        value=""
      />
      <button
        class="bg-black text-white text-xs px-10 py-4"
        type="submit"
      >
        SUBSCRIBE
      </button>
    </form>
  </div>
</div>
`;
